astropy/io/votable/util.py

```diff
@@ -169,8 +207,7 @@ def is_frame_of_reference(x):
         length = len(points)
         if has_frame_of_reference:
             if frames is not None and p[-1] not in frames:
-                raise ValueError(
-                    f"'{p[-1]}' is not a valid frame of reference")
+                raise ValueError(f"{p[-1]!r} is not a valid frame of reference")
```

Comment:
I missed this in the initial suggestions.  I won't object to it, but I'm curious of the rationale.  If the intent of the formatting was to wrap that value in single quotes in the message, then I find wrapping it in single quotes like `'{p[-1]}'` quite intuitive, and the `!r` somewhat cryptic and less indicative of the intent.  Is there a more general guidance or concern I'm missing?  For more complex objects, I suppose using the `repr()` could give different content, but these just look like strings.

Reply:
@nstarman requested this change. I am not sure what his reason was however.

Reply:
Oh. Unless someone has written a custom `__repr__` implementation on an object (not the case here, I think), then ``{x!r}`` is equivalent and preferred to ``'{x}'``.

Reply:
We're just leveraging the Format mini-language.

Reply:
But this is not strictly part of the `black` PR, but something I wish pyupgrade could fix.

Reply:
I agree that in the absence of a custom `__repr__`, `{x!r}` is functionally equivalent to `'{x}'`, but not sure why it would be preferred.  I prefer the latter because to me it more explicitly shows the intent and doesn't require familiarity with Format syntax minutia, but it's not a strong preference, so I'm fine either way here.  I was just curious if it went beyond that, for example if it were considered important to get the custom `__repr__` value if it existed.

Reply:
I think it's because f"{x}" actually calls the ``__str__`` method, which is meant to be overridden more than ``__repr__`` (see https://docs.python.org/3/reference/datamodel.html?highlight=__str__#object.__str__ for expectations). The single quotes indicate ``repr`` was intended, which is why I switched it. If we *want* to use the string form, then that's alright too, just maybe best to lose the surrounding quotes.

Reply:
Ah, thanks for explaining that.  I see what you're saying.  

I interpreted the quotes differently though.   That is, not seeing them as wanting `__repr__` but just wanting to wrap that value in quotes for the user.  With that interpretation, I do prefer the original implementation using the quotes with the `__str__` value, but again I don't think it matters much.